src/workers/AGENTS.md

```diff
@@ -69,15 +71,50 @@ import { MatchedCandidate } from '@dailydotdev/schema';
 
 const worker: TypedWorker<'gondul.v1.candidate-opportunity-match'> = {
   subscription: 'api.store-candidate-opportunity-match',
-  handler: async ({ data }, con): Promise<void> => {
-    // data is typed as MatchedCandidate
+  handler: async (message, con): Promise<void> => {
+    // message.data is typed as MatchedCandidate
+    const { data } = message;
   },
   parseMessage: (message) => {
     return MatchedCandidate.fromBinary(message.data);
   },
 };
 ```
 
+### Typed Notification Workers (`TypedNotificationWorker`)
+
+For notification-specific workers that return `NotificationHandlerReturn`:
+
+```typescript
+import { TypedNotificationWorker } from './worker';
+
+const worker: TypedNotificationWorker<'post-upvoted'> = {
+  subscription: 'api.article-upvote-milestone-notification',
+  handler: async (data, con, logger) => {
+    // Returns NotificationHandlerReturn for notification processing
+    return { type: NotificationType.ArticleUpvoteMilestone, ... };
+  },
+};
+```
+
+### Experiment Workers (`ExperimentWorker`)
+
+For workers that need access to the GrowthBook experiment allocation client:
+
+```typescript
+import { ExperimentWorker, workerToExperimentWorker } from './worker';
+
+const worker: ExperimentWorker = {
+  subscription: 'api.experiment-allocated',
+  handler: async (message, con, logger, pubsub, experimentAllocationClient) => {
```

Comment:
make sure it calls `experimentAllocationClient.waitForSend` before exiting the worker logic so that allocation client sends the allocation to GB